lib/ViewModels/ToolsPanelViewModel.js

```diff
@@ -257,25 +281,11 @@ function getAllRequests(types, mode, requests, group, promises, blacklistGroup)
             }
         } else if ((types.indexOf(item.type) !== -1) && (mode !== 'enabled' || item.isEnabled)) {
 
-
             if (!alreadyInRequests(item)) {
-                var enabledHere = false;
-                if (!item.isEnabled) {
-                    enabledHere = true;
-                    item._enable();
-                }
-                var imageryProvider;
-                if (defined(item._imageryLayer)) {
-                    imageryProvider = item._imageryLayer.imageryProvider;
-                    promises.push(whenImageryProviderIsReady(imageryProvider));
-                }
-                requests.push({
-                    item : item,
-                    blacklistGroup : blacklistGroup,
-                    group : group.name,
-                    enabledHere : enabledHere,
-                    provider : imageryProvider
-                });
+                    //this provides closure for the for loop inside the promise chain
+                var promise = when(item._load()).then( ( generateReguest ) (item, group, blacklistGroup) );
```

Comment:
Did you try this with `load` instead of `_load`?  I think `load` is better (assuming there isn't some non-obvious problem with it) so that it won't load more than once and will set `isLoading` in the process.  Basically, use the public API unless there's a reason we can't.